### DISCUSSION TITLE
Beginnerproblems in VisualStudio and Anaconda

### DISCUSSION THREAD

**Discussion by toben123** [Textualize/textual#3995] [Q&A] [answered]
Hi everyone,
i want to develop self-contained GUIs/TUIs, Textual just looks great for that. I have some beginners problems.

1) If i try to run the examples from the website, Anaconda tells me "RunttimeError: asyncio.run() cannot be called from a running event loop" which from what i learned has something to do with how anaconda runs its commands

so i switched to Visual Studio, when i do that and run the "text-input" example from the Website

2) I get some weird output and it seems to print something about every second

This is kind of discouraging - am i doing something wrong or running into beginnerproblems? I wasnt able to find solutions to my problems and its kind of odd that even the simple examples dont work, even though "python -m textual" works just fine

**Comment by davep** [comment]
Do you think you could start by telling us what platform you're working on, and then also giving us some idea what this "weird output" looks like?

**Comment by toben123** [comment]
Im using Windows with:
Conda Version 23.3.1 with Jupyter Notebook 6.5.2 with Python 3.10.9

And Visual Studio:
Visual Studio 2022 with Python 3.11 64 Bit

i also uploaded some weird output for the text-input example from the website

![IMG_1808](https://github.com/Textualize/textual/assets/58788032/88323bc2-b273-47f2-91fb-a653789e30c4)

**Reply by davep** [reply, marked as answer]
Sadly it's almost impossible for me to see what's going on in that image.

You mention "Jupyter Notebook", which you didn't mention earlier. Textual doesn't currently support applications being run in that environment; Textual applications are designed to be run in a terminal emulator (although we do aim to make Jupyter support work in the future; some work has been done in that direction).

Which version of Windows are we talking about here?

I also just realised that when you say "Visual Studio" you mean the full version, not Visual Studio Code. I can't personally speak for full Visual Studio, but for this to work the terminal emulator built into it would need to have full ANSI support, much like a normal terminal emulator would have. Do you happen to know if it does?

More generally, the terminals provided by some IDEs tend to be quite cut-down and not a great place to run terminal-based applications in general (think of it as trying to test a full-featured website with all the bells and whistles, in a text-only web browser, and expecting to see all the fancy bells and whistles).

If the Visual Studio builtin terminal isn't capable (and from the image you post that would be my guess -- while I can't read the text it looks like it's probably a lot of unhandled ANSI escape sequences), you may find you have a better time with Visual Studio Code (it's terminal is very capable), or follow the [advice for Windows in the "Your Platform" box in the getting started guide in our docs](https://textual.textualize.io/getting_started/).

**Comment by toben123** [comment]
Its Windows10. Anyhow, with VisualStudioCode it works fine just fine with output in the embedded terminal. Thank you very much, i will start using that then. Sorry for being inprecise with my infos.

**Reply by davep** [reply]
Do note that, if I recall correctly, some mouse support in the VSCode terminal is lacking when you run on Windows 10 (this is a Windows/VSCode issue), but on the whole you should find it works well.

**Reply by toben123** [reply]
OK, thank you very much for that further info. Which Platform/Setup would you want me to use ideally so its all just smooth sailing?
